lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1895,6 +1896,37 @@ class DecomposeAtenLeakyReluBackwardOp
 };
 } // namespace
 
+namespace {
+class DecomposeAtenLerpScalarOp : public OpRewritePattern<AtenLerpScalarOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenLerpScalarOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto resType = op.getType().cast<ValueTensorType>();
+    if (!resType.hasDtype()) {
+      return rewriter.notifyMatchFailure(op, "result should have dtype");
+    }
+    if (!op.getType()
+             .cast<ValueTensorType>()
+             .getDtype()
+             .isa<mlir::FloatType>()) {
+      op.emitError("unimplemented: non-floating point dtype");
+      return failure();
+    }
```

Comment:
You don't need to constraint this to the floating point.

Reply:
@vivekkhandelwal1 Should I refactor this to be a check if complex type is result type or should I remove this check completely?

Reply:
I don't see any constraints on the result type in the [documentation for this op](https://pytorch.org/docs/stable/generated/torch.lerp.html), so no need to have this check.